backend/models/user.js

```diff
@@ -0,0 +1,36 @@
+import mongoose from "mongoose";
+import uniqueValidator from "mongoose-unique-validator";
+
+const { Schema } = mongoose;
+
+
+// Template for defining schema 
+// ID is handled by the DB automatically
+const userSchema = new Schema(
+    {
+        username: {
+            type: String,
+            required: true,
+            unique: true,
+        },
+        name: String,
+        passwordHash: String,
+        genrePrefs: [{
+            type: String
+        }]
+    }
+);
+
+userSchema.plugin(uniqueValidator)
+userSchema.set('toJSON', {
+	transform: (document, returnedObject)=>{
+		returnedObject.id = returnedObject._id.toString()
+		delete returnedObject._id
+		delete returnedObject.__v
+		delete returnedObject.passwordHash
+	}
+})
+
+const User = mongoose.model('User', userSchema)
```

Comment:
const User = mongoose.model('User', userSchema, 'users')

'users' connects to the collection on the cloud